infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -258,97 +221,64 @@
     @Configurable(description = "Your budget for this node source related Azure resources. Will be used to compute your global cost in % budget.", sectionSelector = 2)
     protected double maxBudget = 50;
 
+    @Configurable(textArea = true, description = "Linux VM startup script to launch the ProActive nodes (optional). Please refer to the documentation for full description. (optional)", sectionSelector = 8)
+    protected String linuxStartupScript = initScriptGenerator.getDefaultLinuxStartupScript();
+
+    @Configurable(textArea = true, description = "Powershell script to be run during Windows VM startup for launching the ProActive nodes (optional). Please refer to the documentation for full description. (optional)", sectionSelector = 8)
+    protected String windowsStartupScript = initScriptGenerator.getDefaultWindowsStartupScript();
```

Comment:
Couldn't the windowsStartup field be implemented in other infra connectors ? Since you have implemented a provision script generator for windows environment, maybe this could be of use in other cloud providers as well.

Reply:
Other infra connectors use jclouds, since jclouds only supports to run the startup script through ssh connection, so they don't support windows os by default.